File: 240409/홀수 짝수에 따른 출력값/output-value-based-on-odd-even-numbers.py
N = int(input())

def print_sum(n):
    if n == 0 or n == 1:
        return n

    if n % 2 == 0:
        return n + print_sum(n - 2)
    else:
        return n + print_sum(n - 2)

print(print_sum(N))

File: 251226/홀수 짝수에 따른 출력값/output-value-based-on-odd-even-numbers.cpp
#include <iostream>

using namespace std;

int N;
int print_sum(int num);

int main() 
{
    cin >> N;

    // Please write your code here.
    cout << print_sum(N);
    return 0;
}

int print_sum(int num)
{
    if(num == 0 || num == 1)
        return num;

    if(num % 2 == 0)
        return num + print_sum(num - 2);
    else
        return num + print_sum(num - 2);
}